server/opts.go

```diff
@@ -1678,11 +1720,86 @@ func parseExportStreamOrService(v interface{}, errors, warnings *[]error) (*expo
 				*errors = append(*errors, err)
 			}
 		}
-
 	}
 	return curStream, curService, nil
 }
 
+// parseServiceLatency returns a latency config block.
+func parseServiceLatency(root token, v interface{}) (*serviceLatency, error) {
+	if subject, ok := v.(string); ok {
+		return &serviceLatency{
+			subject:  subject,
+			sampling: DEFAULT_SERVICE_LATENCY_SAMPLING,
+		}, nil
+	}
+
+	latency, ok := v.(map[string]interface{})
+	if !ok {
+		return nil, &configErr{token: root,
+			reason: fmt.Sprintf("Expected latency entry to be a map/struct, got %T", v)}
+	}
+
+	sl := serviceLatency{
+		sampling: DEFAULT_SERVICE_LATENCY_SAMPLING,
+	}
+
+	// Read sampling value.
+	if v, ok := latency["sampling"]; ok {
+		tok, ok := v.(token)
+		if !ok {
+			return nil, &configErr{token: root,
+				reason: fmt.Sprintf("Expected latency token, got %T", tok)}
+		}
+
+		var sample int64
+		switch v := tok.Value().(type) {
```

Comment:
Is v set to type here va value?

Reply:
I cleaned up some of the config and handling tokens here to report line correctly, should be fixed now.